pyproject.toml

```diff
@@ -66,6 +66,8 @@ dependencies=[
   "ruamel.yaml",
   "sympy",
   "tabulate",
+  # Keep this version in sync with: third-party/ao pin
+  "torchao==0.10.0",
```

Comment:
In main, how would you depend on latest ao changes? I'm worried that if you bump the hash in third-party/ao (that may not have the a dedicated release) and it becomes inconsistent and confusing.

Here's what I suggest

In main: 

- Only update third-party/ao to a hash that is consistent with 0.10.0. But not update pyproject.toml

This is not 100% necessary but it's good because (1) we're testing the CI in main (2) making sure main's hash is not older than release branch's

In release branch branch:

- update third-party/ao to a hash that is consistent with 0.10.0. 
- update pyproject.toml to be 0.10.0
- update the release checklist to reflect this new process about pinning to torchao.